lib/http/2/compressor.rb

```diff
@@ -549,7 +550,16 @@ def header(buf)
       # @return [Array] +[[name, value], ...]+
       def decode(buf)
         list = []
-        list << @cc.process(header(buf)) until buf.empty?
+        decoding_pseudo_headers = true
+        until buf.empty?
+          next_header = @cc.process(header(buf))
+          is_pseudo_header = next_header.first.start_with? ':'
+          if !decoding_pseudo_headers && is_pseudo_header
+            fail ProtocolError, 'one or more pseudo headers encountered after regular headers'
```

Comment:
Hmm, one gotcha here..

https://github.com/igrigorik/http-2/blob/master/lib/http/2/connection.rb#L609-L611

^ we'll end up raising compression error regardless. I think we need to update that logic as well.

Reply:
I have updated the logic in `connection.rb`.  Pushing momentarily.